### ISSUE TITLE
Bug: Data leakage — scaler/encoder fit_transform() called on full dataset before train/test split

### ISSUE THREAD

**Issue by Daksh7785** [SdSarthak/AegisAI#514]
## Bug Report

**Description**
In `Automated Financial Reporting with Deep Learning/app.py`, the `MinMaxScaler` is fitted on
the **entire dataset** (including what would be test data) before any train/test split. This
causes **data leakage** — the model indirectly sees test data statistics during training,
leading to artificially inflated accuracy metrics that won't hold in real-world use.

## Affected Files

| File | Line | Issue |
|------|------|-------|
| `Automated Financial Reporting with Deep Learning/app.py` | 16 | `scaler.fit_transform()` on entire df |
| `Engineering Career Role Prediction/app.py` | 32 | `le.fit_transform()` inside prediction function using training encoder on live input |

## Code (Buggy)
```python
scaler = MinMaxScaler()
data_scaled = scaler.fit_transform(df.drop('Date', axis=1))  # leaks test stats into training
```

## Fix
```python
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
scaler = MinMaxScaler()
X_train_scaled = scaler.fit_transform(X_train)   # fit ONLY on train
X_test_scaled = scaler.transform(X_test)          # transform test separately
```

## Severity
High — produces misleadingly optimistic model performance; this is a teaching repo so it
actively trains beginners with incorrect ML practices

**Comment by snowlindhanusha** [comment]
Hi, I am a GSSoC’26 contributor. I would like to work on this issue. Please assign it to me.

**Comment by akhila-sraj** [comment]
Hi @Daksh7785 , I am a GSSoC 2026 contributor and would like to work on this issue.

I have experience with cybersecurity concepts and secure application development, and I’d love to contribute. Please assign this to me.

**Comment by madsysharma** [comment]
Hi @SdSarthak , I would like to work on this as part of GSSoC 2026.

**Comment by Abhushan187** [comment]
Hi @SdSarthak, I'm a GSSoC '26 contributor and would like to fix the data leakage issue in app.py by ensuring scaler/encoder fit_transform() is called only on training data after the train/test split. Please assign this to me!

**Comment by Daksh7785** [comment]
I'd like to work on this — claiming it.
